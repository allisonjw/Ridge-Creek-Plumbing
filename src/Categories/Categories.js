import React from 'react';
import './Categories.scss';
import Projects from '../Projects/Projects';
import '../Portfolio/Portfolio.scss';
import Portfolio from '../Portfolio/Portfolio';

export const Categories = ({ pickedType, type, allProjectsArray }) => {

  if (type === 'all') {
    const allCategoriesArrayList = allProjectsArray.map(project => <Projects key={project.id} {...project} />);
    return (
      <>
          <Portfolio allCats={allCategoriesArrayList} />
      </>
    )
   } else {
    const projectList = pickedType.map(project => <Projects key={project.id} {...project} />);
      return (
        <>
            <Portfolio projectList={projectList} />
        </>
      )
    };
  };    
 
export default Categories;