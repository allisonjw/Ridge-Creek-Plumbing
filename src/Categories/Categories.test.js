import React from 'react';
import Categories from './Categories';
import { shallow } from 'enzyme';

describe('Categories', () => {
    it('should match the snapshot if there is a project type', () => {
        let type = '';
        let pickedType = ['commercial'];
        let allProjectsArray = [{
            id: 1,
            project_type: 'commercial',
            name: 'Puyallup School District',
            location: 'Puyallup, WA',
            image: require('../images/puyallup.jpg'),
        },
        {
            id: 2,
            project_type: 'commercial',
            name: 'The Artisans Group',
            location: 'South Sound Region',
            image: require('../images/artisans.jpg'),
        }];

        const projectWrapper = shallow(
            <Categories 
                type={type}
                pickedType={pickedType}
                allProjectsArray={allProjectsArray}
            />
        );
        expect(projectWrapper).toMatchSnapshot();
    });

    it('should match the snapshot if the type is all', () => {
        let type = 'all';
        let pickedType= [' '];
        let allProjectsArray = [{
            id: 1,
            project_type: 'commercial',
            name: 'Puyallup School District',
            location: 'Puyallup, WA',
            image: require('../images/puyallup.jpg'),
        },
        {
            id: 2,
            project_type: 'commercial',
            name: 'The Artisans Group',
            location: 'South Sound Region',
            image: require('../images/artisans.jpg'),
        }];

        const projectWrapper = shallow(
            <Categories 
                type={type}
                pickedType={pickedType}
                allProjectsArray={allProjectsArray}
            />
        );
        expect(projectWrapper).toMatchSnapshot();
    });
});