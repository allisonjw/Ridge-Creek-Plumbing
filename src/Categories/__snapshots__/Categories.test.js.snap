// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories should match the snapshot if the type is all 1`] = `ShallowWrapper {}`;

exports[`Categories should match the snapshot if there is a project type 1`] = `ShallowWrapper {}`;
